File: backend/src/main/scala/microservices/hotel/objects/GetHotelDetailsPlannerRequest.scala
package com.typesafe.travel.hotel.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class GetHotelDetailsPlannerRequest(
    hotelId: String,
    checkInDate: Option[String],
    checkOutDate: Option[String]
)

object GetHotelDetailsPlannerRequest:
  given sourceEncoder: Encoder[GetHotelDetailsPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[GetHotelDetailsPlannerRequest] = deriveDecoder

File: frontend/src/microservices/hotel/objects/GetHotelDetailsPlannerRequest.ts
export type GetHotelDetailsPlannerRequest = {
  hotelId: string
  checkInDate?: string
  checkOutDate?: string
}

export const getHotelDetailsPlannerRequestFromJson = (json: string): GetHotelDetailsPlannerRequest =>
  JSON.parse(json) as GetHotelDetailsPlannerRequest

export const getHotelDetailsPlannerRequestToJson = (value: GetHotelDetailsPlannerRequest): string =>
  JSON.stringify(value)
